### PR TITLE
[pipelining] Rename ManualPipelineStage -> PipelineStage

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128201
* __->__ #128157

Renaming ManualPipelineStage to remove the "Manual" part. I needed to replace the existing `PipelineStage` which takes in the `pipe` argument, so I have renamed that to `TracerPipelineStage`. @kwen2501 will remove this entirely in favor of adding a util to `Pipe` to just create the stage directly.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Review by wconstab — APPROVED** [review]
lgtm

File: docs/source/distributed.pipelining.rst [hunk 1]
```diff
@@ -156,7 +156,7 @@ First off, let us instantiate a ``PipelineStage`` instance:
   # Pipeline stage is our main pipeline runtime. It takes in the pipe object,
   # the rank of this process, and the device.
   from torch.distributed.pipelining import PipelineStage
-  stage = PipelineStage(pipe, rank, device)
+  stage = TracerPipelineStage(pipe, rank, device)
```
Comment (H-Huang) [inline docs/source/distributed.pipelining.rst:159]: will wait for https://github.com/pytorch/pytorch/pull/128070 to land so this doesnt cause a merge conflict

**Pull request closed without merging**